mobile/src/main/java/org/openhab/habdroid/model/ServerProperties.java

```diff
@@ -0,0 +1,176 @@
+    public static final int SERVER_FLAG_SSE_SUPPORT   = 1 << 1;
+
+    public static class UpdateHandle {
+        public void cancel() {
+            if (call != null) {
+                call.cancel();
+                call = null;
+            }
+        }
+
+        private Call call;
+        private Builder builder;
+    }
+
+    public interface UpdateSuccessCallback {
+        void handleServerPropertyUpdate(ServerProperties props);
+    }
+    public interface UpdateFailureCallback {
+        void handleUpdateFailure(Request request, int statusCode, Throwable error);
+    }
+
+    public abstract int flags();
+    public abstract List<OpenHABSitemap> sitemaps();
+
+    public boolean hasJsonApi() {
+        return (flags() & SERVER_FLAG_JSON_REST_API) != 0;
+    }
+
+    public boolean hasSseSupport() {
+        return (flags() & SERVER_FLAG_SSE_SUPPORT) != 0;
+    }
+
+    abstract Builder toBuilder();
+
+    @AutoValue.Builder
+    static abstract class Builder {
+        abstract Builder flags(int flags);
+        abstract Builder sitemaps(List<OpenHABSitemap> sitemaps);
+
+        abstract ServerProperties build();
+        abstract int flags();
+    }
+
+    public static UpdateHandle updateSitemaps(ServerProperties props, Connection connection,
+            UpdateSuccessCallback successCb, UpdateFailureCallback failureCb) {
+        UpdateHandle handle = new UpdateHandle();
+        handle.builder = props.toBuilder();
+        fetchSitemaps(connection.getAsyncHttpClient(), handle, successCb, failureCb);
+        return handle;
+    }
+
+    public static UpdateHandle fetch(Connection connection,
+            UpdateSuccessCallback successCb, UpdateFailureCallback failureCb) {
+        final UpdateHandle handle = new UpdateHandle();
+        handle.builder = new AutoValue_ServerProperties.Builder();
+        fetchFlags(connection.getAsyncHttpClient(), handle, successCb, failureCb);
+        return handle;
+    }
+
+    private static void fetchFlags(AsyncHttpClient client, UpdateHandle handle,
+            UpdateSuccessCallback successCb, UpdateFailureCallback failureCb) {
+        handle.call = client.get("/rest", new AsyncHttpClient.StringResponseHandler() {
+            @Override
+            public void onFailure(Request request, int statusCode, Throwable error) {
+                failureCb.handleUpdateFailure(request, statusCode, error);
+            }
+
+            @Override
+            public void onSuccess(String response, Headers headers) {
+                try {
+                    JSONObject result = new JSONObject(response);
+                    // If this succeeded, we're talking to OH2
+                    int flags = SERVER_FLAG_JSON_REST_API;
+                    client.removeHeader("Accept");
+                    try {
+                        String versionString = result.getString("version");
+                        int versionNumber = Integer.parseInt(versionString);
```

Comment:
Demo server and my local servers cause a nfe here.
My test server (2.?-SNAPSHOT) returns `0.10.0.201712141004`, demo server `0.10.0.b1`.

Reply:
A 2.3 snapshot is needed for it to work. See https://github.com/eclipse/smarthome/pull/5032 for the numbering change.